### PR TITLE
[IR][copy] prevent loose of location information for non inlinable lambdas

### PR DESCRIPTION
**Pull request by vvlevchenko**

### CODE REVIEW

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt [hunk 1]
```diff
@@ -65,6 +65,14 @@ open class DeepCopyIrTreeWithSymbols(
     }
 
     protected fun mapDeclarationOrigin(origin: IrDeclarationOrigin) = origin
+    protected fun mapDeclarationOrigin(declaration: IrDeclaration):IrDeclarationOrigin {
```
Comment (homuroll) [inline compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt:68]: Won't it be enough to override this method in the inliner copier?

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt [hunk 2]
```diff
@@ -65,6 +65,14 @@ open class DeepCopyIrTreeWithSymbols(
...
+        return when {
+            declaration.origin == IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA && (declaration is IrFunction) && !declaration.isInline -> {
```
Comment (homuroll) [inline compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt:70]: Do really such functions exist - with origin = LOCAL_FUNCTION_FOR_LAMBDA and inline?

Comment (vvlevchenko) [reply compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt:70]: Hm... good question. will investigate it.

File: compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt (hunk 1 above)
Comment (vvlevchenko) [reply compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DeepCopyIrTreeWithSymbols.kt:68]: will try

**Comment by vvlevchenko** [comment]
replaced with #3604

**Pull request closed without merging**